File: training/go/domain/model/armor_class.go
package model

type ArmorClass struct {
	Description string
	Base        int
	Bonus       int
}

func NewArmorClass(base int, bonus int) ArmorClass {
	return ArmorClass{Description: "Natural Armor", Base: base, Bonus: bonus}
}

func (ac ArmorClass) Total() int {
	return ac.Base + ac.Bonus
}

File: sample-projects/csharp/src/De.Sots.CellarsAndCentaurs/Domain/Model/ArmorClass.cs
using De.Sots.CellarsAndCentaurs.Application;

namespace De.Sots.CellarsAndCentaurs.Domain.Model;

public sealed class ArmorClass
{
    public int Base { get; private set; }

    public int Bonus { get; private set; }

    public string Description { get; }

    public int Total => Base + Bonus;

    public ArmorClass(int baseValue, int bonus, string? description = null)
    {
        Base = baseValue;
        Bonus = bonus;
        Description = description ?? CreatureUtil.STANDARD_ARMOR_CLASS_DESCRIPTION;
    }

    public void Improve(int bonus)
    {
        Bonus += bonus;
    }
}
